nexus_constructor/main_window.py

```diff
@@ -129,7 +129,9 @@ def _update_views(self):
     def _update_3d_view_with_component_shapes(self):
         for component in self.model.entry.instrument.component_list:
             shape, positions = component.shape
-            self.sceneWidget.add_component(component.name, shape, positions)
+            self.sceneWidget.add_component(
+                component.name, component.nx_class, shape, positions
```

Comment:
as python, excluding primitive types, sends everything by reference, we could just have add_component(component: Component) and extract everything we need in add_component. In that way we do not need to update everything everytime we need something from the Component class. What do you think?

Reply:
Fix in later PR.